index.html

```diff
@@ -132,6 +133,9 @@
 						}
 					}
 				}
+				map.setLayoutProperty(sourceID + '-poly', 'visibility', 'none');
```

Comment:
The way I've made deemphasising the other districts work is that the House or Senate districts polygon layer is now effectively a mask.  There's a potentially confusing sequence of events:

1. On first load, it's technically `visible` but has the 0 alpha value that is set in the `addVectorLayer()` call that first loads it.  This is necessary because it needs to be visible _to mapbox-gl.js_ for my code to be able to find each district's bounding box to set the zoom control up, and leaving it as fully transparent prevents that from being visible _to the user_.
2. Once the `populateZoomControl()` function has finished stepping through all the districts, it sets the polygon layer's `visibility` property to `none`, and then the next two lines set up the colour for the mask - currently a translucent light grey.  This is now safe to do because the layer isn't visible.
3. Then when we zoom to a district, the `zoomToPolygon()` function makes this mask layer visible again, but filters out the one district we're interested in.  That way everything else is behind the translucent mask.

The main practical consequence of this is that if you want to change the appearance of the mask, you should do that by editing the `map.setPaintProperty()` calls here, _not_ the properties with which it's initially loaded, because those will get overridden by whatever's specified here.

Reply:
That's pretty slick! And pretty tricky. Thanks for the detailed explanation of what's happening, I have a feeling it will come in useful down the road.

Reply:
Indeed!  I'm patting myself on the back for this because it's going to save me reinventing the wheel to add this to the pre-K map.  I probably won't have time to finish it today, though.